src/components/ScheduleSetup.tsx

```diff
@@ -231,75 +261,62 @@ export default class ScheduleSetup extends React.Component<ScheduleSetupProps, S
-                  onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
-                    props.messages[index].text = event.target.value;
-                    props.onMessagesChanged(props.messages);
-                    // this.setState({messages: this.state.messages});
-                  }}
-                />
-              </Grid>
+    const buildMessageItem = (message: CommunicationRequest, index: number) => {
+        return <ListItem key={index} sx={{
+            width: '100%'
+        }} secondaryAction={
+            message.status === "completed" ? <></> :
+            <IconButton hidden={message.status === "completed"} onClick={() => {
+                removeMessage(index);
+            }}>
+                <ClearIcon/>
+            </IconButton>
+        }>
+            <Grid container direction={"row"} flexDirection={"row"} spacing={2}
+                  sx={{paddingTop: 2}}>
+
+                <Grid item>
+                    <DateTimePicker
+                        label={message.status === "completed" ? "Delivered Date & Time" : "Scheduled Date & Time"}
+                        value={message.occurrenceDateTime}
+                        inputFormat="ddd, MM/DD/YYYY hh:mm A" // example output display: Thu, 03/09/2023 09:34 AM
+                        disabled={message.status === "completed"}
+                        onChange={(newValue: Date | null) => {
+                            message.setOccurrenceDate(newValue);
+                            props.onMessagePlanChanged(props.messagePlan);
+                        }}
+                        renderInput={(params: TextFieldProps) => <TextField {...params} sx={{ width: "264px" }} />}
+                    />
+                </Grid>
+                <Grid item flexGrow={1}>
+                    <TextField
+                        error={message.getText().length === 0}
+                        helperText={message.getText().length === 0 ? "Enter a message" : ""}
+                        label={message.status === "completed" ? "Delivered Message" : "Scheduled Message"}
+                        fullWidth
+                        multiline
+                        value={message.getText() ?? ""}
+                        placeholder={"Enter message"}
+                        disabled={message.status === "completed"}
+                        onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
+                            message.setText(event.target.value);
+                            props.onMessagePlanChanged(props.messagePlan);
+                        }}/>
+                </Grid>
             </Grid>
-          </ListItem>
-        );
+        </ListItem>
     }
 
     const removeMessage = (index: number) => {
-        let messages = props.messages;
-        messages.splice(index, 1);
-        props.onMessagesChanged(messages);
+        let messagePlan = props.messagePlan;
+        messagePlan.removeActiveCommunicationRequest(index);
```

Comment:
@hannahburkhardt hmm.. looks like I am no longer able to remove a message. 
[see here](https://user-images.githubusercontent.com/12942714/224865537-fcd4172e-bbcc-4c42-935f-a2419824f419.webm)
I was able to in the [shared dev instance](https://femr.isacc.dev.cirg.uw.edu/home)

Reply:
I _think_ the issue might be [here](https://github.com/uwcirg/isacc-messaging-client-sof/pull/12/files#diff-63ce71e720747f893a3baa1e98f3b9c83cdc625c9ae455f5aea8c1aac4669f95R127)

Reply:
Thanks! fixed.